src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -669,6 +687,33 @@ TER DirectStepI::check (StrandContext const& ctx) const
         }
     }
 
+    // The previous checks apply in all cases.  The following checks apply
+    // only if we're not offer crossing because they require a preexisting
+    // trustline.
+    if (offerCrossing_)
```

Comment:
Let's say I want to place an offer to acquire IOU/GATEWAY, an asset for which I do not have a trust line. My offer  goes on the books. At some point in the future, GATEWAY adds `lsfRequireAuth` and then, a couple of minutes later, my offer is crossed.

I shouldn't be able to acquire any funds, because I'm not authorized to hold GATEWAY assets; where, exactly, do we check for that?